src/channel.ts

```diff
@@ -1135,27 +1149,12 @@ export class Channel<
    * @return {number} Unread count
    */
   countUnread(lastRead?: Date | Immutable.ImmutableDate | null) {
-    if (!lastRead) {
-      return this.state.unreadCount;
-    }
+    if (!lastRead) return this.state.unreadCount;
 
     let count = 0;
-    for (const m of this.state.messages.asMutable()) {
-      const message = m.asMutable({ deep: true });
-      if (this.getClient().userID === message.user?.id) {
-        continue;
-      }
-      if (m.shadowed) {
-        continue;
-      }
-      if (m.silent) {
-        continue;
-      }
-      if (lastRead == null) {
-        count++;
-        continue;
-      }
-      if (m.created_at > lastRead) {
+    for (let i = 0; i < this.state.messages.length; i += 1) {
```

Comment:
caching length might help

Reply:
`i++` looks more idiomatic

Reply:
the performance difference is negligible for caching the length
https://eslint.org/docs/rules/no-plusplus

Reply:
makes sense, we can assume this is resolved.